RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -0,0 +1,148 @@
+            .Throws<InvalidOperationException>();
+
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
+        });
+
+        response.StatusCode.ShouldBe(HttpStatusCode.InternalServerError);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenBrewStatusUnavailable_ShouldReturnEmpty503()
+    {
+        var cache = new MemoryCache(new MemoryCacheOptions());
+        long numRequestsSoFar = 4;
+        cache.Set(CoffeeMachineController.CACHE_KEY_NUM_REQUESTS, numRequestsSoFar);
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IMemoryCache>(cache);
+        });
+
+        response.StatusCode.ShouldBe(HttpStatusCode.ServiceUnavailable);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Theory]
+    [InlineData(0, StatusCodes.Status200OK)]
+    [InlineData(1, StatusCodes.Status200OK)]
+    [InlineData(2, StatusCodes.Status200OK)]
+    [InlineData(3, StatusCodes.Status200OK)]
+    [InlineData(4, StatusCodes.Status503ServiceUnavailable)]
+    [InlineData(5, StatusCodes.Status200OK)]
+    public async Task GetBrewStatus_WhenEndpointCalled_ShouldReturnCorrectStatusCode(long numRequestsSoFar, int expectedStatusCode)
+    {
+        var cache = new MemoryCache(new MemoryCacheOptions());
+        cache.Set(CoffeeMachineController.CACHE_KEY_NUM_REQUESTS, numRequestsSoFar);
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IMemoryCache>(cache);
+        });
+
+        response.ShouldNotBeNull();
+        ((int)response.StatusCode).ShouldBe(expectedStatusCode);
+    }
+
+    [Theory]
+    [InlineData(0)]
+    [InlineData(4)]
+    public async Task GetBrewStatus_WhenApril1st_ShouldReturnEmpty418(long numRequestsSoFar)
+    {
+        var cache = new MemoryCache(new MemoryCacheOptions());
+        cache.Set(CoffeeMachineController.CACHE_KEY_NUM_REQUESTS, numRequestsSoFar);
+
+        var dateTimeProvider = new Mock<IDateTimeProvider>();
+        dateTimeProvider.Setup(p => p.GetLocalNow())
+            .Returns(new DateTime(2023, 4, 1));
+
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IMemoryCache>(cache);
+            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
+        });
+
+        ((int)response.StatusCode).ShouldBe(StatusCodes.Status418ImATeapot);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenCoffeeReady_ShouldReturn200()
+    {
+        var nowLocal = new DateTime(2023, 1, 2, 3, 4, 5, DateTimeKind.Local);
```

Comment:
A bit odd to use local time in a test case (it works because both the test case & application use operating-system time).

This is a requirement that remained ambiguous & I made an assumption that operating-system time could be used, & this is further highlighted in a code comment in the api project.